scalableminds/webknossos
# Add support for transformations with thin plate splines

Adds support for a thin-plate-spline transform which is defined as a list of correspondences (source and target points). An (inverse) TPS is found which fits these correspondences. The resulting weights are passed to the shader which then uses these to transform the coordinates ad-hoc.
The rest of the code (mostly the bucket pickers) don't use the TPS, but instead use an affine approximation which is also derived from the correspondences. The downside of this approach is that the bucket picking can get inaccurate if the affine approximation deviates from the actual TPS transformation. However, in the one known demo case, this is not an issue, which is why I'd keep it as is for now.

### Steps to test:
- open this dataset: https://tps.webknossos.xyz/datasets/sample_organization/C555_DIAMOND_2f-TPS/view#513,512,480,0,0.424
- ensure correct rendering
- you can compare it with https://tps.webknossos.xyz/datasets/sample_organization/C555_DIAMOND_2f/view#513,512,480,0,0.424 which is the same DS but with an affine-only transformation

### Issues:
- fixes #6965 

------
(Please delete unneeded items, merge only when none are left open)
